### PR TITLE
Add cache, rows stats and connection errors data to status metricset in MySQL module

### PR DESCRIPTION
**Pull request by sayden**
## What does this PR do?
Adds cache, connection errors and rows stats metrics to the  `status` Metricset in MySQL module as requested in https://github.com/elastic/beats/issues/16955

I'll have to confirm metrics in this PR by the team before attaching any dashboard to avoid modifying it multiple times.

## Why is it important?
The enhancement is a step forward to improve the MySQL module.

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have made corresponding change to the default configuration files
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.

## Author's Checklist

- [ ] Dashboard contains new metrics

## Related issues

- Implements / Closes https://github.com/elastic/beats/issues/16955

### CODE REVIEW

File: metricbeat/module/mysql/status/_meta/fields.yml [hunk 1]
```diff
@@ -264,10 +333,26 @@
...
       type: group
       description: >
       fields:
+      - name: rows
+        type: group
+        description: >
+        fields:
+          - name: reads
+            type: long
+            description: The number of rows reads into InnoDB tables.
```
Comment (ChrsMark) [inline metricbeat/module/mysql/status/_meta/fields.yml:346]: maybe:
```suggestion
            description: The number of rows reads from InnoDB tables.
```

File: metricbeat/module/mysql/status/_meta/fields.yml [hunk 2]
```diff
@@ -264,10 +333,26 @@
...
+          - name: inserted
+            type: long
+            description: The number of rows inserted into InnoDB tables.
+          - name: deleted
+            type: long
+            description: The number of rows deleted into InnoDB tables.
```
Comment (ChrsMark) [inline metricbeat/module/mysql/status/_meta/fields.yml:352]: maybe:
```suggestion
            description: The number of rows deleted from InnoDB tables.
```

File: metricbeat/module/mysql/status/data.go [hunk 3]
```diff
@@ -125,6 +125,36 @@ var (
 				},
 				"write_requests": c.Int("Innodb_buffer_pool_write_requests"),
 			},
+			"rows": s.Object{
```
Comment (ChrsMark) [inline metricbeat/module/mysql/status/data.go:128]: Maybe we need to add some Unit tests in the future for this module so as to secure this schema won't break?
Sth similar to https://github.com/elastic/beats/blob/b5220465832b305f1ed3ae12838c4dd4fa345314/metricbeat/module/nats/connections/connections_test.go#L32

Comment (sayden) [reply metricbeat/module/mysql/status/data.go:128]: Yeap, a full set of tests would be helpful and not just the ones that check the parent key. I have created an issue for it https://github.com/elastic/beats/issues/19880

**Comment by sayden** [comment]
CI failures are related, checking

**Pull request merged**